StarTrek13/Xel/borg_structures.dm

```diff
@@ -199,55 +201,168 @@
+	desc = "A borg assembly line. It fabricates parts at immense speeds."
+	icon_state = "factory"
+	bound_width = 96
+	layer = 5.5
+	var/running
+	var/part_time = 20 //DEBUG. currently set for two seconds PER.//per part.
+	var/required_points = 0 // How many assimilation points to construct this item?
+	var/constructables = list(/obj/item/borg/manip, /obj/item/borg/amp) //What can this factory build"?
+	//var/construct_amount = 4 //How many per cycle?
+	var/constructing = null
+	var/saved_time
+
+/obj/machinery/borg/factory/Destroy()
+	//STOP_PROCESSING(SSobj, src)
+	construction(TRUE)
+	..()
+
+/obj/machinery/borg/factory/update_icon()
+	if(running)
+		icon_state = "factory_on"
 	else
-		src.say("there is already one of those here")
-		qdel(src)
+		icon_state = "factory"
+
+/obj/machinery/borg/factory/attack_hand(mob/living/carbon/human/user)
+	if(!isborg(user))
+		return
+	var/query
+	query = input(user, "Select an item for production.","Select Production",query) as anything in constructables|"cancel production" // Yes, I know this lists the item paths, not names. It's temporary.
+	if(!query)
+		return
+	if(SSticker.mode.hivemind.cons_points < required_points)
+		to_chat(user, "<span class='warning'>We do not have enough construction materials to fabricate this part.</span>")
+		return
+	if(query == "cancel production")
+		construction(TRUE)
+		update_icon()
+		return
+	to_chat(user, "<span class='info'>The [src] is now producing [query].</span>")
+	constructing = query
+	//part_time = query.cons_time * 10
+	saved_time = world.time + part_time
+	construction()
+	update_icon()
+
+/obj/machinery/borg/factory/proc/construction(var/stop)
+	to_chat(world, "PROCESSING A BORG FACTORY")
+	if(stop)
+		running = FALSE
+		to_chat(world, "PROCESSING A BORG FACTORY STOPPED")
+		//update_icon()
+		return
+	running = TRUE
+	//update_icon()
+	while(world.time < saved_time)
+		sleep(1)
+		to_chat(world, "PROCESSING A BORG FACTORY WHILE")
+	saved_time = world.time + part_time
+	if(!SSticker.mode.hivemind.use_cons_points(required_points))
+		//STOP_PROCESSING(SSobj, src)
+		to_chat(world, "A BORG FACTORY IS MISSING MATERIALS")
+		running = FALSE
+		SSticker.mode.hivemind.message_collective("The [src] at [get_area(src)] has automatically cancelled production. Construction reserves depleted.")
+		update_icon()
+		return
+	new constructing(src.loc)
+	to_chat(world, "BORG FACTORY MADE AN ITEM")
+	construction()
+
+/obj/machinery/borg/converter
+	name = "conversion device"
+	desc = "The final stage of the assimilation process of the borg. May god have mercy on your crew."
+	icon_state = "converter"
+	bound_width = 96
+	layer = 5.5
+	var/running
+	var/inserted_parts = list()
+	var/required_parts = list()
+
+/obj/machinery/borg/converter/update_icon()
+	if(running)
+		icon_state = "converter_active"
+	else
+		icon_state = "converter"
+
+/obj/machinery/borg/converter/attack_hand(mob/living/carbon/human/user) // TODO: MULTISTRUCTURE REQUIREMENT
```

Comment:
this entire proc gave me multiple terminal diseases

Reply:
You're quite welcome.

Reply:
find a better way tio do all this, every time you do a for each loop it gets worse